apps/wallet-mobile/src/Search/SearchContext.tsx

```diff
@@ -113,7 +121,13 @@ export const useSearchOnNavBar = ({
      *   1) go back when the search input is not visible
      *   2) close the search input when the search input is visible
      */
-    if (!visible) navigation.goBack()
+    if (visible) return
+
+    if (navigateBack) {
```

Comment:
I believe this will cause the navigation to never go back to the receiver when there is at least one amount added.